tools/mason/MasonLogger.chpl

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020-2025 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module MasonLogger {
+  import IO;
+  import this.SafeCalls as Safe;
+
+
+  // Note: we use >= to determine whether we should output something. So, if you
+  // change enum values, make sure to put things in order, where the lower
+  // values mean lower verbosity.
+  enum logLevel { no, error, warn, info, debug };
+
+  config const logs = logLevel.info;
```

Comment:
since mason uses arg parsing, this should probably be made an arg, not a config const for better ergonomics.

If this works today, I'd be happy with just a todo comment

Reply:
I added a comment. I think this should be discussed under the same umbrella as what to do with `--show` as `--show` is using argumentParser. Where I am not really against --show at all. I just want to be able to control how verbose mason is globally (`--show` is per-command) with varying levels of output including completely quiet. Maybe we have the logger as I have today with 5 different levels, but also `--quiet` and `--show` (or, I mean, `--verbose`??) referring to 2 of those levels as sugar and no additional implementation to mess with.